### ISSUE TITLE
AttributeError: 'SearchPlugin' object has no attribute 'is_dirty'

### ISSUE THREAD

**Issue by vedranmiletic** [squidfunk/mkdocs-material#6364]
### Bug description

If one enables Material theme after running `mkdocs serve`, the build will fail with:

```
% mkdocs serve 
INFO    -  Building documentation...
INFO    -  Cleaning site directory
INFO    -  Documentation built in 0.18 seconds
INFO    -  [17:53:32] Watching paths for changes: 'docs', 'mkdocs.yml'
INFO    -  [17:53:32] Serving on http://127.0.0.1:8000/
INFO    -  [17:53:36] Browser connected: https://staging.group.miletic.net/
INFO    -  [17:53:47] Detected file changes
INFO    -  Building documentation...
Traceback (most recent call last):
  File "/home/vedranm/software/mkdocs/.venv/lib/python3.9/site-packages/mkdocs/livereload/__init__.py", line 193, in _build_loop
    func()
  File "/home/vedranm/software/mkdocs/.venv/lib/python3.9/site-packages/mkdocs/commands/serve.py", line 67, in builder
    build(config, live_server=None if is_clean else server, dirty=is_dirty)
  File "/home/vedranm/software/mkdocs/.venv/lib/python3.9/site-packages/mkdocs/commands/build.py", line 354, in build
    config.plugins.on_post_build(config=config)
  File "/home/vedranm/software/mkdocs/.venv/lib/python3.9/site-packages/mkdocs/plugins.py", line 542, in on_post_build
    return self.run_event('post_build', config=config)
  File "/home/vedranm/software/mkdocs/.venv/lib/python3.9/site-packages/mkdocs/plugins.py", line 509, in run_event
    result = method(**kwargs)
  File "/home/vedranm/software/mkdocs/.venv/lib/python3.9/site-packages/material/plugins/search/plugin.py", line 136, in on_post_build
    if self.is_dirty:
AttributeError: 'SearchPlugin' object has no attribute 'is_dirty'
ERROR   -  [17:53:49] An error happened during the rebuild. The server will appear stuck until build errors are resolved.
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Built-in search plugin](https://squidfunk.github.io/mkdocs-material/plugins/search/)

### Reproduction

[9.4.10-searchplugin-is_dirty.zip](https://github.com/squidfunk/mkdocs-material/files/13403906/9.4.10-searchplugin-is_dirty.zip)

### Steps to reproduce

1. `mkdocs new mysite`
2. `cd mysite`
3. `mkdocs serve`
4. Add the following lines to `mkdocs.yml` while `mkdocs serve` is running:
    ``` yaml
    theme:
      name: material
    ```
5. Notice the `AttributeError` in the terminal.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. This is really a minor bug, and can be mitigated by just restarting the server. However, we'll see whether we can fix it.

**Comment by vedranmiletic** [comment]
Thanks, minor indeed, no rush at all. I just randomly encountered it while generating the minimal reproduction for #6365, which is the bug we actually care about much more.

**Comment by squidfunk** [comment]
Fixed in ad83798be. That was easy.

**Comment by vedranmiletic** [comment]
Thank you!

**Comment by TimChaubet-I4U** [comment]
My apologies for being inconsistent. What I meant with my [post here](https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsquidfunk%2Fmkdocs-material%2Fissues%2F5686%23issuecomment-1821067228&data=05%7C01%7Ctim.chaubet%40dedalus.eu%7Cc382bf65fc104e9444c508dbeaa0c748%7C9ffff5c3bdfa4a9db595ff68329945ef%7C0%7C0%7C638361748480061542%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=fFCyzRdJLAwzo44zcornJjdrx73Ktgf9I%2BZbMqTdJnI%3D&reserved=0) is we still need to use `mkdocs build` to get the search plugin activated. I read here that this is fixed. My guess is that you haven't rebuilt the docker image yet.

The docker container runs mkdocs in serve mode, so we need to launch a `docker compose exec -it mkdocs sh -c 'mkdocs build'` command when we enable a plugin, like search. Just adding it to the yaml file is not enough.
We've added a rebuild command to the startup command of the container, so that restarting it does a build, then serve. So we have a workaround.

From what I understood from this bugreport being fixed, is that `mkdocs serve` would, from now on, do a build if it's necessary to get a plugin working, when it's added to the yml, but that was perhaps wishful thinking.

**Comment by squidfunk** [comment]
Correct, the fix is still unreleased. If an issue is tagged as _resolved_, but still open, it is unreleased. We'll release it in the coming days, which will also rebuild the Docker image. Since this is a very minor bug, there's no hurry from our side.

**Comment by squidfunk** [comment]
Released as part of [9.4.11](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.11).
